lib/marklogic.js

```diff
@@ -788,7 +789,9 @@ function initClient(client, inputParams) {
     client.request = http.request;
     if (noAgent) {
       mlutil.copyProperties(inputParams, agentOptions, [
-        'keepAliveMsecs', 'maxSockets', 'maxFreeSockets', 'timeout'
+        'ca', 'cert', 'ciphers', 'keepAliveMsecs', 'key', 'maxSockets',
+        'maxFreeSockets', 'passphrase', 'pfx', 'rejectUnauthorized',
+        'secureProtocol', 'timeout', 'maxCachedSessions'
```

Comment:
As stated in the issue, I believe that the agent options for non-SSL HTTP should be:

keepAliveMsecs, maxFreeSockets, maxSockets, maxTotalSockets, scheduling, timeout